### ISSUE TITLE
KeyValue request fails on non-ASCII key

### ISSUE THREAD

**Issue by jongwook** [couchbase/couchbase-jvm-core#5]
[`com.couchbase.client.core.endpoint.kv.KeyValueHandler` sets memcached's key length bytes](https://github.com/couchbase/couchbase-jvm-core/blob/1.1.2/src/main/java/com/couchbase/client/core/endpoint/kv/KeyValueHandler.java#L203-L208) by calling `String.length()`, which equals to the length of the actual key blob sent to the server only if the key is always ASCII.

For example, the following will result in `RequestCancelledException: Request cancelled in-flight.`

```scala
import com.couchbase.client.java.CouchbaseCluster
import com.couchbase.client.java.document.JsonDocument
import com.couchbase.client.java.document.json.JsonObject

object CouchbaseTest extends App {

  val cb = CouchbaseCluster.create("guile072")
  val bucket = cb.openBucket("default")
  val key = "키"
  println(bucket.upsert(JsonDocument.create(key, JsonObject.fromJson("""{"key":"value"}"""))))
  println(bucket.get(key))
  
}
```

In the above example a non-ASCII key `"키"` is used, whose UTF-8 binary is 3 bytes whereas the keyLength is encoded as 1. The above code works perfectly for keys like `"key"`.

Below are the bytes sent to TCP; you can see that the key length (bytes 2-3) and the total body length (bytes 8-11) are set incorrectly: should be (`00 03` and `00 00 00 1a` respectively)

```
16:08:16.375 [cb-io-1-8] TRACE c.c.c.d.i.n.h.logging.LoggingHandler - [id: 0x08e481cd] WRITE(35B)
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 80 01 00 01 08 00 02 75 00 00 00 18 00 00 00 01 |.......u........|
|00000010| 00 00 00 00 00 00 00 00 02 00 00 00 00 00 00 00 |................|
|00000020| ed 82 a4                                        |...             |
+--------+-------------------------------------------------+----------------+
16:08:16.375 [cb-io-1-8] TRACE c.c.c.d.i.n.h.logging.LoggingHandler - [id: 0x08e481cd] WRITE(15B)
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 7b 22 6b 65 79 22 3a 22 76 61 6c 75 65 22 7d    |{"key":"value"} |
+--------+-------------------------------------------------+----------------+
```

**Comment by daschl** [comment]
@jongwook this is fixed in 2.1.3!

**Comment by jongwook** [comment]
Thanks for letting me know, will close this issue. JVMCBC-187
